rfcs/FINEGRAINED_NOTIFICATIONS.md

```diff
@@ -0,0 +1,122 @@
+# Finegrained notifications
+
+## Summary
+
+This document describes an API proposal for a finegrained collection for Realm Java. Object-level notifications will be covered in another document.
+
+## Motivation
+
+Realm wants to promote a reactive pattern to working with your data. It should be possible to define a query for any change you are interested in and just depend on Realm figuring out when that change has happend and notify you. 
+
+This is interesting for a number of use cases:
+
+1) I want to animate or refresh a RecyclerView when elements are added, moved, modified or deleted in a `RealmResults`.
```

Comment:
use `.` instead of `)` to make it look better 🍡 
1. I want to animate or refresh a RecyclerView when elements are added, moved, modified or deleted in a `RealmResults`.
2. I want to be notified about the result of an query changing so I can refresh my UI.
3. I want my changelisteners to _only_ trigger when there is a relevant change.